bhttp/src/http_compat/encode.rs

```diff
@@ -0,0 +1,441 @@
+
+                // 5. Change to next state
+                this.state = HttpEncodeState::BodyChunk;
+                Poll::Ready(Some(Ok(buf)))
+            }
+            HttpEncodeState::BodyChunk => {
+                // Poll the body for more data
+                let body = this.message.body_mut();
+                match Pin::new(body).poll_frame(cx) {
+                    Poll::Ready(Some(Ok(frame))) => {
+                        if let Some(data) = frame.data_ref() {
+                            // Is a data frame
+                            let mut chunk_data = Vec::new();
+                            write_vec(&data, &mut chunk_data)?;
+                            Poll::Ready(Some(Ok(chunk_data)))
+                        } else if let Some(trailers) = frame.trailers_ref() {
+                            // Is a trailers frame
+                            let mut buf = Vec::new();
+                            // First, we need to write zero-length chunk to indicate end, since we always using Mode::IndeterminateLength
+                            write_len(0, &mut buf)?;
+
+                            // Then, write the trailer field section with indeterminate length mode
+                            let field_section = FieldSection::try_from(trailers)?;
+                            field_section.write_bhttp(Mode::IndeterminateLength, &mut buf)?;
+
+                            // Switch to Done state since we have written the trailers
+                            this.state = HttpEncodeState::Done;
+                            Poll::Ready(Some(Ok(buf)))
+                        } else {
+                            // For unknown frame types, report an invalid data error.
+                            this.state = HttpEncodeState::Done;
+                            Poll::Ready(Some(Err(Error::UnknownHttpBodyFrameType)))
+                        }
+                    }
+                    Poll::Ready(Some(Err(e))) => Poll::Ready(Some(Err(Error::Io(
+                        std::io::Error::new(std::io::ErrorKind::Other, e),
+                    )))),
+                    Poll::Ready(None) => {
+                        // End of body, write zero-length chunk to indicate end
+                        let mut end_chunk = Vec::new();
+                        write_len(0, &mut end_chunk)?;
+                        this.state = HttpEncodeState::Done;
+                        Poll::Ready(Some(Ok(end_chunk)))
+                    }
+                    Poll::Pending => Poll::Pending,
+                }
+            }
+            HttpEncodeState::Done => Poll::Ready(None),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use bytes::Bytes;
+    use futures::executor::block_on;
+    use futures::StreamExt;
+    use http::{Request, Response};
+    use http_body::Frame;
+    use http_body_util::Full;
+    use http_body_util::StreamBody;
+
+    #[test]
+    fn test_request_to_bhttp_conversion() {
```

Comment:
I should enable clippy::redundant_test_prefix:

```suggestion
fn request_to_bhttp_conversion() {
```